templates/cc/register.go

```diff
@@ -201,12 +201,12 @@ func (fns CCFuncs) errIdxCause(ctx shared.RuleContext, idx, cause string, reason
 			errName, fns.lit(pgsgo.PGGUpperCamelCase(f.Name()))))
 
 	if idx != "" {
-		output = append(output, fmt.Sprintf(`msg << "[" << %s << "]";`, fns.lit(idx)))
+		output = append(output, fmt.Sprintf(`msg << "[" << %s << "]";`, idx))
 	} else if ctx.Index != "" {
-		output = append(output, fmt.Sprintf(`msg << "[" << %s << "]";`, fns.lit(ctx.Index)))
+		output = append(output, fmt.Sprintf(`msg << "[" << %s << "]";`, ctx.Index))
 	}
 
-	output = append(output, fmt.Sprintf(`msg << ": " << %s;`, fns.lit(fmt.Sprintf("%q", reason))))
+	output = append(output, fmt.Sprintf(`msg << ": " << %s;`, fmt.Sprintf("%q", fmt.Sprint(reason...))))
```

Comment:
The triple Sprint looks suspicious. Why is this necessary?

Reply:
It isn't necessary. I managed to simplify it. I was following the implementation in `templates/goshared/register.go` but accidentally over-complicated it. Thanks for the feedback.

Here's the [goshared](https://github.com/envoyproxy/protoc-gen-validate/blob/main/templates/goshared/register.go#L128) implementation for reference:
```go
return fmt.Sprintf(`%s{
		field: %s,
		reason: %q,
		%s%s
	}`,
		fns.errName(f.Message()),
		fld,
		fmt.Sprint(reason...),
		causeFld,
		keyFld)
```

Reply:
I'm not a Go expert but it looks like you should be able to unwrap `fmt.Sprint(reason...)` as just `reason...` since there's a single format parameter.

Reply:
I am also not a Go expert (rookie would be more appropriate), but it seems `fmt.Sprint(reason...)` is necessary in this case. `reason...` can contain any number of different types of data (basically it's a slice of templates, e.g. `[]interface{}`). `fmt.Sprint(reason...)` will unpack the elements, convert them to their appropriate string representation, and concatenate them. The `%q` in the outer `fmt.Sprintf` will then wrap this string in double quotes and properly escape the string.

I tried what you suggested but `fmt.Sprintf` doesn't accept varargs the same way `fmt.Sprint` does, so I get output that looks like this:
```
msg << ": " << "value must be ";%!(EXTRA uint64=3, string= characters)
```

If you have any other suggestions I'd be happy to try them. But I (with my limit Go experience) don't see a better way right now.

Reply:
Thanks for the explanation and for trying it out.